src/core/Group.cpp

```diff
@@ -1059,13 +1059,7 @@ Entry* Group::addEntryWithPath(const QString& entryPath)
 
 void Group::applyGroupIconTo(Entry* entry)
 {
-    if (!config()->get("UseGroupIconOnEntryCreation").toBool()) {
-        return;
-    }
-
-    if (iconNumber() == Group::DefaultIconNumber && iconUuid().isNull()) {
```

Comment:
This conditional is important because it prevents setting the blue folder icon on entries.